test/unit/teacher_set_test.rb

```diff
@@ -182,13 +193,293 @@ class TeacherSetTest < ActiveSupport::TestCase
+            resp = TeacherSet.delete_teacher_set(bib_id)
+          end
+        end
+      end
+      assert_equal(teacher_set.bnumber, resp.bnumber)
+      @mintest_mock1.verify
+    end
+
+    it 'test bib record not found exception' do
+      bib_id = '1234'
+      resp = nil
+      teacher_set = TeacherSet.new(bnumber: "")
+      @mintest_mock1.expect(:call, [], [bib_id])
+      resp = assert_raises(BibRecordNotFoundException) do
+        TeacherSet.stub :get_teacher_set_by_bnumber, @mintest_mock1 do
+          teacher_set.stub :delete_teacherset_record_from_es, teacher_set do
+            resp = TeacherSet.delete_teacher_set(bib_id)
+          end
+        end
+      end
+      assert_equal(BIB_RECORD_NOT_FOUND[:code], resp.code)
+      assert_equal(BIB_RECORD_NOT_FOUND[:msg], resp.message)
+      @mintest_mock1.verify
+    end
+  end
+
   def test_teacher_set_query
     params = {"page"=>"1", "controller"=>"teacher_sets", "action"=>"index", "format"=>"json"}
     resp = TeacherSet.for_query(params)
-    assert_equal(4, resp.count)
+    assert_equal(5, resp.count)
+  end
+
+
+  # Update teacher-set document in ES
+  describe '#test create or update teacherset' do
+    before do
+      @es_doc = {"_index" => "teacherset", "_type" => "teacherset", 
+                "_id" => "353", "_version" => 11, "result" => "noop", 
+                "_shards" => {"total" => 0, "successful" => 0, "failed" => 0}}
+      
+      created_at = Time.zone.parse("2014-10-19 1:00:00")
+      updated_at = Time.zone.parse("2014-10-19 1:00:00")
+      @ts_obj = TeacherSet.new(title: "title", description: "description", contents: "contents", 
+      id: 123, details_url: "details_url", grade_end: 1, 
+      grade_begin: 1, availability: "available", total_copies: 1,
+      call_number: 12345, language: "English", physical_description: "physical_description",
+      primary_language: "English", created_at: created_at, updated_at: updated_at,
+      available_copies: 1, bnumber: "B456", set_type: "set_type",
+      area_of_study: "area_of_study", subjects: [])
+
+      @expected_resp = {:title => "title",:description => "description",:contents => "contents",
+       :id => 123,:details_url => "details_url",:grade_end => 1,:grade_begin => 1,:availability => "available",
+       :total_copies => 1, :call_number => "12345",:language => "English",
+       :physical_description => "physical_description",:primary_language => "English",
+       :created_at => "2014-10-19T01:00:00+0000",:updated_at => "2014-10-19T01:00:00+0000",:available_copies => 1,
+       :bnumber => "B456",:set_type => "set_type",:area_of_study => "area_of_study",:subjects => []}
+    end
+
+    it 'update teacher-set document in es' do
+      resp = nil
+      @mintest_mock1.expect(:call, @expected_resp)
+      @mintest_mock2.expect(:update_document_by_id, @es_doc, [@expected_resp[:id], @expected_resp])
+      
+      @ts_obj.stub :teacher_set_info, @mintest_mock1 do
```

Comment:
Similarly, I don't think we need stub `teacher_set_info` — once you've created @ts_obj you should be able to just call it. It does make sense to mock `update_document_by_id` on ElasticSearch though — that looks good!